nautobot_ssot/contrib/base.py

```diff
@@ -0,0 +1,47 @@
+"""Abstract base classes for primary contrib module classes.
+
+These base classes are meant to define the expected interactions between the various classes
+within contrib to allow for additional customization and extension based on individual needs.
+
+Todo:
+  - Add abstract & common methods
+  - Update references in adapter and model
+"""
```

Comment:
I like the idea. I would like to see some real examples of customization that is made possible by this pattern and how it is not possible before.